firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -271,6 +275,14 @@ String getName() {
     return Tasks.call(backgroundExecutor, this::deleteFirebaseInstallationId);
   }
 
+  /** Register a callback {@link FidListener} to receive fid changes. */
+  @Override
+  public void registerFidListener(@Nullable FidListener listener) {
+    synchronized (this) {
```

Comment:
it looks equivalent to just making the whole method `saynchronized`, please consider doing that.